### ISSUE TITLE
Time zone is not respected when using DateTime.to_string

### ISSUE THREAD

**Issue by Munksgaard** [elixir-cldr/cldr_dates_times#52]
I tried the following in rapid succession:

```iex
iex [09:22 :: 36] > Cldr.DateTime.to_string!(DateTime.utc_now(), time_zone: "UTC")
"Oct 7, 2024, 2:16:08 PM"
iex [09:22 :: 37] > Cldr.DateTime.to_string!(DateTime.utc_now(), time_zone: "Europe/Copenhagen")
"Oct 7, 2024, 2:16:10 PM"
```

I would've expected the time_zone to affect the datetime printed, but that doesn't seem to actually happen. It's also unclear from the documentation what format is expected for the time_zone argument.

**Comment by kipcole9** [comment]
@Munksgaard, `Cldr.DateTime.to_string/2` takes a `:format` option that describes how the date time is formatted. By default it is `:medium` which a locale-independent way of expressing a format. There is also `:short" and `:long`. You can see what those formats are, for a given locale, with `Cldr.DateTime.available_formats`. For example:

```elixir
iex> Cldr.DateTime.formats(:da)
{:ok,
 %Cldr.DateTime.Formats{
   short: "{1} {0}",
   medium: "{1} {0}",
   long: "{1} {0}",
   full: "{1} {0}"
 }}
```
Now I realise that's not very illuminating, You would need to know that this is a format where the date is formatted, the time is formatted and the results are substituted into `{0}` and `{1}` accordingly. It also means that `Cldr.Date.to_string/2` and `Cldr.Time.to_string/2` are called to produce those results.

So we can see what those formats are:
```elixir
iex> Cldr.Date.formats(:da)
{:ok,
 %Cldr.Date.Formats{
   short: "dd.MM.y",
   medium: "d. MMM y",
   long: "d. MMMM y",
   full: "EEEE 'den' d. MMMM y"
 }}
iex> Cldr.Time.formats(:da)
{:ok,
 %Cldr.Time.Formats{
   short: "HH.mm",
   medium: "HH.mm.ss",
   long: "HH.mm.ss z",
   full: "HH.mm.ss zzzz"
 }}
```

And you can see that `format: :medium`, for locale `:da` (and many/most locales) does not format the timezone. But `format: :long` and `format: :full` do.

You can also specify other locale-independent formats by looking at:
```elixir
iex`> Cldr.DateTime.available_formats(:da)
{:ok,
 %{
   yQQQ: "QQQ y",
   EBhm: "E h.mm B",
   Bhms: "h.mm.ss B",
   Ehm: "E h.mm a",
   GyMMMd: "d. MMM y G",
   Hm: "HH.mm",
   yw: %{other: "'uge' w 'i' Y", pluralize: :week_of_year, one: "'uge' w 'i' Y"},
   MMdd: "dd.MM",
   h: "h a",
   hm: "h.mm a",
   hmsv: "h.mm.ss a v",
   yMd: "d.M.y",
   Ehms: "E h.mm.ss a",
   y: "y",
   yMMMM: "MMMM y",
   d: "d.",
   Gy: "y G",
   EBhms: "E h.mm.ss B",
   Hmsv: "HH.mm.ss v",
   E: "ccc",
   M: "M",
   yMEd: "E d.M.y",
   EHms: "E HH.mm.ss",
   yQQQQ: "QQQQ y",
   MEd: "E d.M",
   Ed: "E 'den' d.",
   yMMM: "MMM y",
   GyMd: "d.M.y GGGGG",
   MMMEd: "E d. MMM",
   ms: "mm.ss",
   MMMMEd: "E d. MMMM",
   GyMMMEd: "E d. MMM y G",
   yMM: "MM.y",
   Md: "d.M",
   yMMMEd: "E d. MMM y",
   Bh: "h B",
   Bhm: "h.mm B",
   hmv: "h.mm a v",
   hms: "h.mm.ss a",
   yM: "M.y",
   MMMd: "d. MMM",
   Hmv: "HH.mm v",
   GyMMM: "MMM y G",
   MMMMd: "d. MMMM",
   H: "HH",
   MMMMW: %{other: "W. 'uge' 'i' MMMM", pluralize: :week_of_month, ...},
   Hms: "HH.mm.ss",
   MMM: "MMM",
   ...
 }}
```
And interestingly, none of those formats include the timezone.

Lastly, there is no `:timezone` option to `Cldr.DateTime.to_string.2` and I can't see a reference to it in the docs for that function.

Did I indicate there is such an option somewhere? I'll need to fix that.

Does that help?

**Comment by Munksgaard** [comment]
> And interestingly, none of those formats include the timezone.

I apologize, but I don't think I communicated my desires properly: I am not interested in showing the timezone for a give DateTime, but I am interested in taking a DateTime and shift the time to e.g. the time zone used by the user of my app. In other words, at 13:00 UTC today, I expected `Cldr.DateTime.to_string!(DateTime.utc_now(), time_zone: "Europe/Copenhagen")` to return the string `"Oct 7, 2024, 3:00:00 PM"`.

> Lastly, there is no :timezone option to Cldr.DateTime.to_string.2 and I can't see a reference to it in the docs for that function.

I see references to `:time_zone` (note the underscore) both [here](https://hexdocs.pm/ex_cldr_dates_times/Cldr.DateTime.html#content), [here](https://hexdocs.pm/ex_cldr_dates_times/Cldr.DateTime.html#to_string/3) and [here](https://hexdocs.pm/ex_cldr_dates_times/Cldr.DateTime.html#to_string!/3), but I realize now that this was in describing the expected format of the `DateTime` argument to e.g. `to_string/3`, and not to any options of `to_string/3` itself.

Really, I just want to show timestamps for my users in their local time zone (e.g. one of my users was confused that they created an entity and immediately it said that it was created 2 hours in the past, because it shows the UTC time instead of the CET time). Upon closer inspection, I suspect I'll have to work with [`DateTime.shift_zone/3`](https://hexdocs.pm/elixir/main/DateTime.html#module-converting-between-timezones) before I hand off the DateTime to Cldr?

**Comment by kipcole9** [comment]
>  I just want to show timestamps for my users in their local time zone
Yes, `DateTime.shift_zone/3` is the right tool I believe. `ex_cldr_dates_times` is a formatter, not a general date time library.

> I see references to :time_zone (note the underscore) both [here](https://hexdocs.pm/ex_cldr_dates_times/Cldr.DateTime.html#content), [here](https://hexdocs.pm/ex_cldr_dates_times/Cldr.DateTime.html#to_string/3) and [here](https://hexdocs.pm/ex_cldr_dates_times/Cldr.DateTime.html#to_string!/3)

I will work on improving the docs to remove any ambiguity.

**Comment by Munksgaard** [comment]
> I will work on improving the docs to remove any ambiguity.

I don't think that's necessary. You should be able to expect that people read the documentation you've written, I was just in too much of a hurry :-)
